openstack/networking/v2/extensions/lbaas_v2/pools/requests.go

```diff
@@ -110,15 +116,33 @@ type CreateOpts struct {
 // Create accepts a CreateOpts struct and uses the values to create a new
 // load balancer pool.
 func Create(c *gophercloud.ServiceClient, opts CreateOpts) CreateResult {
+	var res CreateResult
+
+	// Validate inputs
```

Comment:
(personal style comment, feel free to ignore: Rather than optimise for early failure detection, I'd leave it up to the server-side to validate arguments like these.  The server will return effectively the same error you're generating here, and it allows for clients+servers to evolve new supported options over time without having to extend the hardcoded list in gophercloud)